### PR TITLE
Manage downtimes with dst (recurrent and realtime)

### PR DESCRIPTION
**Pull request by kduret**
to cherry-pick on master
An other pull request is coming for 2.7.x

### CODE REVIEW

File: www/class/centreonExternalCommand.class.php [hunk 1]
```diff
@@ -400,6 +400,43 @@ public function acknowledgeService(
      * Downtime
      ***********/
 
+    private function getDowntimeTimestampFromDate($date = 'now', $timezone = '', $start = true)
+    {
+        $dateTime = new \DateTime($date, new \DateTimeZone($timezone));
+
+        // Winter to summer dst
+        $dateTime2 = clone($dateTime);
```
Comment (loiclau) [inline www/class/centreonExternalCommand.class.php]: clone $dateTime  without incidentally

File: www/class/centreonDowntime.Broker.class.php [hunk 2]
```diff
@@ -292,6 +292,20 @@ private function manageWinterToSummerTimestamp($time, $timestamp, $timezone)
         return $timestamp;
     }
 
+    private function manageSummerToWinterTimestamp($timestamp, $timezone)
+    {
+        $dstDate = new DateTime('now', $timezone);
+        $dstDate->setTimestamp($timestamp);
+        $dateTime2 = clone($dstDate);
```
Comment (loiclau) [inline www/class/centreonDowntime.Broker.class.php]: clone

File: features/bootstrap/DowntimeDSTContext.php [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        $this->spin(
+            function ($context) {
+                $scheduled = false;
+                $return = $context->container->execute(
+                    "cat /var/log/centreon-engine/centengine.log",
+                    'web'
+                );
+                $output = $return['output'];
+                if (
+                    preg_match_all(
```
Comment (MatthieuMan) [inline features/bootstrap/DowntimeDSTContext.php]: Styles: phpcs mark this new line as an error

Comment (MatthieuMan) [reply features/bootstrap/DowntimeDSTContext.php]: 290

File: features/bootstrap/DowntimeDSTContext.php [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        $this->spin(
+            function ($context) {
+                $scheduled = true;
+                $return = $context->container->execute(
+                    "cat /var/log/centreon-engine/centengine.log",
+                    'web'
+                );
+                $output = $return['output'];
+                if (
+                preg_match(
```
Comment (MatthieuMan) [inline features/bootstrap/DowntimeDSTContext.php]: Styles: phpcs mark this new line as an error (290)

**Pull request merged**